### PR TITLE
add geodataframe table driver

### LINKED ISSUE #924: GeoDataFrame `"vector"` driver
### Kind of request

Changing existing functionality

### Enhancement Description

In v0.x, there is a "vector" driver for `GeoDataFrame`s. This driver uses [`open_vector_from_table`](https://github.com/Deltares/hydromt/blob/v0.9.4/hydromt/io.py#L663). This needs to be included in v1.

### Use case

Open GeoDataFrames that are saved in table formats.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Jaapel**
## Issue addressed
Fixes #924 

## Explanation
Created a "table" driver for the previously called "vector" driver.

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation
- [x] Updated changelog.rst

### CODE REVIEW

**Review by savente93 — APPROVED** [review]
LGTM, nicely done

File: hydromt/drivers/geodataframe/table_driver.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+from hydromt._typing.error import NoDataStrategy
+from hydromt._typing.metadata import SourceMetadata
+from hydromt._utils.unused_kwargs import warn_on_unused_kwargs
+from hydromt.drivers.geodataframe.geodataframe_driver import GeoDataFrameDriver
+
+logger: Logger = getLogger(__name__)
+
+# possible labels for the x and y dimensions
+X_DIM_LABELS = ("x", "longitude", "lon", "long")
+Y_DIM_LABELS = ("y", "latitude", "lat")
```
[the remark this answers is not in the document]
Comment (Jaapel) [reply hydromt/drivers/geodataframe/table_driver.py:19]: Why not? :) Is a list smaller in working memory / processing time? Never looked into that, as we often use numpy arrays internally if we talk about significant data.

[the remark this answers is not in the document]
Comment (savente93) [reply hydromt/drivers/geodataframe/table_driver.py:19]: Yeah, I think the difference in python is very minimal. I usually default to lists so I was just wondering if you had a reason for preferring tuples out of curiosity.

File: hydromt/drivers/geodataframe/table_driver.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        warn_on_unused_kwargs(
+            self.__class__.__name__,
+            {"mask": mask, "predicate": predicate, "variables": variables},
+            logger=logger,
+        )
+        if len(uris) > 1:
+            raise ValueError(
+                "DataFrame: Reading multiple files with the "
+                f"{self.__class__.__name__} driver is not supported."
+            )
```
[the remark this answers is not in the document]
Comment (Jaapel) [reply hydromt/drivers/geodataframe/table_driver.py:50]: For `GeoDataFrame`s, it is not: https://github.com/Deltares/hydromt/blob/main/hydromt/data_adapter/geodataframe.py#L305

**Pull request merged**